### PR TITLE
refactor(condo): INFRA-634 && INFRA-643 update ticket and contact pages queries

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**

### CODE REVIEW

File: apps/condo/domains/common/utils/next/apollo.ts [hunk 1]
```diff
@@ -49,9 +49,33 @@ const cacheConfig: InitCacheConfig = (cacheOptions) => {
...
+                    timeToLive: 60 * 1000, // 1 minute in milliseconds
+                },
+                TicketStatus: {
+                    timeToLive: 60 * 60 * 1000, // 1 hour in milliseconds
+                },
+                TicketSource: {
+                    timeToLive: 60 * 60 * 1000, // 1 hour in milliseconds
+                },
+                TicketClassifier: {
+                    timeToLive: 60 * 60 * 1000, // 1 hour in milliseconds
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/common/utils/next/apollo.ts:71]: Its should also cached on the back-end side for longer time

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/common/components/Comments/index.tsx [hunk 2]
```diff
@@ -207,34 +215,31 @@ export type CommentWithFiles = TicketComment & {
...
-    // Place for abilities check. If action of given type is not returned, appropriate button will not be displayed
-    actionsFor: (comment: CommentWithFiles) => ActionsForComment
+    createAction: CreateTicketCommentMutationHookResult[0]
+    updateAction: UpdateTicketCommentMutationHookResult[0]
     canCreateComments: boolean
-    refetchComments
+    refetchComments: () => Promise<void>
     FileModel: Module
-    ticketCommentsTime
+    ticketCommentTimes: GetTicketLastCommentsTimeQueryHookResult['data']['ticketCommentTimes'][0]
```
Comment (toplenboren) [inline apps/condo/domains/common/components/Comments/index.tsx:225]: I'd say its best to use get from _ here

File: apps/condo/domains/common/components/Comments/index.tsx [hunk 3]
```diff
@@ -283,10 +290,25 @@ const Comments: React.FC<ICommentsListProps> = ({
...
+                variables: {
+                    data: {
+                        ...values,
+                        type: commentType,
+                        ticket: { connect: { id: ticketId } },
+                        user: { connect: { id: user?.id || null } },
+                    },
+                },
+            })
+            const comment = commentData?.data?.ticketComment
```
Comment (toplenboren) [inline apps/condo/domains/common/components/Comments/index.tsx:311]: best to use get from lodash

File: apps/condo/domains/ticket/components/IncidentHints.tsx [hunk 4]
```diff
@@ -170,7 +173,7 @@ export const IncidentHints: React.FC<IncidentHintsProps> = (props) => {
             },
         })
 
-        return get(res, 'data.objs', [])
+        return res?.data?.incidentProperties?.filter(Boolean) || []
```
Comment (toplenboren) [inline apps/condo/domains/ticket/components/IncidentHints.tsx:176]: best to use lodash get here

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/common/utils/next/apollo.ts [hunk 5]
```diff
@@ -48,6 +48,38 @@ const cacheConfig: InitCacheConfig = (cacheOptions) => {
         },
         invalidationPolicies: {
             timeToLive: 15 * 60 * 1000, // 15 minutes in milliseconds
+            types: {
+                _QueryMeta: {
+                    timeToLive: 60 * 1000, // 1 minute in milliseconds
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/utils/next/apollo.ts:53]: Did you find any solutions to counts?

Comment (SavelevMatthew) [reply apps/condo/domains/common/utils/next/apollo.ts:53]: Since its the most heavy request

File: apps/condo/domains/common/utils/next/apollo.ts [hunk 6]
```diff
@@ -48,6 +48,38 @@ const cacheConfig: InitCacheConfig = (cacheOptions) => {
...
+                    timeToLive: 60 * 60 * 1000, // 1 hour in milliseconds
+                },
+                CallRecord: {
+                    timeToLive: 60 * 1000, // 1 minute in milliseconds
+                },
+                CallRecordFragment: {
+                    timeToLive: 60 * 1000, // 1 minute in milliseconds
+                },
+                Invoice: {
+                    timeToLive: 5 * 60 * 1000, // 5 minutes in milliseconds
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/utils/next/apollo.ts:80]: Why exactly 5?

File: apps/condo/domains/ticket/components/IncidentHints.tsx (hunk 4 above)
Comment (SavelevMatthew) [reply apps/condo/domains/ticket/components/IncidentHints.tsx:176]: Lodash will return you `any` in this case

Comment (SavelevMatthew) [reply apps/condo/domains/ticket/components/IncidentHints.tsx:176]: And in many other cases, so your frontend becomes less type-safe and bug-safe

Comment (SavelevMatthew) [reply apps/condo/domains/ticket/components/IncidentHints.tsx:176]: btw, you can also use `filter(nonNull)` to be even more type-safe

File: apps/condo/domains/ticket/components/TicketForm/TicketFormContext.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ export const TicketFormContextProvider = ({ children, organizationId, isExistedT
 
     const { obj: newTicketSetting, loading: ticketSettingLoading } = TicketSetting.useObject({
         where: { organization: { id: organizationId } },
-    })
+    }, { skip: !organizationId })
```
Comment (SavelevMatthew) [inline apps/condo/domains/ticket/components/TicketForm/TicketFormContext.tsx:29]: persistor?

**Review by SavelevMatthew — APPROVED** [review]

File: apps/condo/domains/contact/components/EditContactForm.tsx [hunk 8]
```diff
@@ -68,41 +74,49 @@ export const EditContactForm: React.FC = () => {
...
-                { organization_is_null: true },
-                { organization: { id: get(contact, 'organization.id', null) } },
-            ],
-        },
+        data: rolesData,
+    } = useGetCommonOrOrganizationContactRolesQuery({
+        variables: { organizationId },
+        skip: !persistor || !organizationId,
     })
+    const roles = rolesData?.roles
```
Comment (Alllex202) [inline apps/condo/domains/contact/components/EditContactForm.tsx:92]: const roles = rolesData?.roles || []

File: apps/condo/domains/organization/queries/OrganizationEmployee.graphql [hunk 9]
```diff
@@ -138,3 +138,26 @@ query getEmployeeInvitesCount($userId: ID!) {
...
+        where: $where,
+        first: 100
+    ) {
+        id
+        name
+        phone
+    }
+}
+
+query getOrganizationEmployeeWithTicketOrganization ($userId: ID!, $organizationId: ID!) {
```
Comment (Alllex202) [inline apps/condo/domains/organization/queries/OrganizationEmployee.graphql:153]: mb getOrganizationEmployeeByTicketOrganization or smth? From title I expect the answer to be organization, it's a bit confusing

File: apps/condo/domains/ticket/contexts/FavoriteTicketsContext.tsx [hunk 10]
```diff
@@ -26,18 +26,23 @@ const useFavoriteTickets = (): IFavoriteTicketsContext => useContext(FavoriteTic
...
         refetch: refetchFavoriteTickets,
         loading,
-    } = UserFavoriteTicket.useAllObjects({
-        where: {
-            user: { id: user.id },
-            ticket: extraTicketsQuery,
+    } = useGetUserFavoriteTicketsQuery({
+        variables: {
+            userId: user.id,
+            ticketWhere: extraTicketsQuery,
```
Comment (Alllex202) [inline apps/condo/domains/ticket/contexts/FavoriteTicketsContext.tsx:38]: wait extraTicketsQuery?

File: apps/condo/pages/contact/[id]/update.tsx [hunk 11]
```diff
@@ -7,9 +7,10 @@ import { PageContent, PageWrapper } from '@condo/domains/common/components/conta
 import { PageComponentType } from '@condo/domains/common/types'
 import { EditContactForm } from '@condo/domains/contact/components/EditContactForm'
 import { ContactsReadAndManagePermissionRequired } from '@condo/domains/contact/components/PageAccess'
+import { prefetchContact } from '@condo/domains/contact/utils/next/Contact'
 
 
-const ContactUpdatePage: PageComponentType = () => {
+const ContactUpdatePage: PageComponentType<{ id: string }> = () => {
```
Comment (Alllex202) [inline apps/condo/pages/contact/[id]/update.tsx:13]: for what?

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/pages/contact/[id]/update.tsx (hunk 11 above)
Comment (nomerdvadcatpyat) [reply apps/condo/pages/contact/[id]/update.tsx:13]: removed

**Pull request closed without merging**